users/tests.py

```diff
@@ -12,39 +13,136 @@ def setUp(self):
         self.client.credentials()
-        self.user_mobile = "+919876543210"
 
     def test_guest_can_request_for_otp(self):
         response = self.client.post(
-            reverse("request-otp"), {"mobile": self.user_mobile}
+            reverse("request-otp"), {"mobile": self.user.mobile}
         )
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
-        otp_exists = OneTimePassword.objects.filter(mobile=self.user_mobile).exists()
+        otp_exists = OneTimePassword.objects.filter(mobile=self.user.mobile).exists()
         self.assertTrue(otp_exists)
 
     def test_invalid_otp_should_fail(self):
         # request otp
-        self.client.post(reverse("request-otp"), {"mobile": self.user_mobile})
+        self.client.post(reverse("request-otp"), {"mobile": self.user.mobile})
 
         # invalid otp
         otp = "000000"
         response = self.client.post(
-            reverse("verify-otp"), {"mobile": self.user_mobile, "otp": otp}
+            reverse("verify-otp"), {"mobile": self.user.mobile, "otp": otp}
         )
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
 
     def test_valid_otp_should_pass(self):
         # request otp
-        self.client.post(reverse("request-otp"), {"mobile": self.user_mobile})
+        self.client.post(reverse("request-otp"), {"mobile": self.user.mobile})
 
         # verify valid otp
-        otp = OneTimePassword.objects.filter(mobile=self.user_mobile).first()
+        otp = OneTimePassword.objects.filter(mobile=self.user.mobile).first()
         response = self.client.post(
-            reverse("verify-otp"), {"mobile": self.user_mobile, "otp": otp.otp}
+            reverse("verify-otp"), {"mobile": self.user.mobile, "otp": otp.otp}
         )
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
+    def test_valid_otp_should_pass_new_user(self):
+        # new user
+        new_user_mobile = "+919988776654"
+
+        # request otp
+        self.client.post(reverse("request-otp"), {"mobile": new_user_mobile})
+
+        # verify valid otp
+        otp = OneTimePassword.objects.filter(mobile=new_user_mobile).first()
+        response = self.client.post(
+            reverse("verify-otp"), {"mobile": new_user_mobile, "otp": otp.otp}
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+class UserTestCase(BaseTestCase):
+    def test_get_config(self):
+        config = {"test": True}
+        # set config
+        self.user.config = config
+        self.user.save()
+
+        # get config
+        response = self.client.get(f"/api/v1/users/{self.user.id}/config/")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json(), config)
+
+    def test_user_cannot_get_other_user_config(self):
+        # get config
+        response = self.client.get(f"/api/v1/users/{self.user_2.id}/config/")
```

Comment:
isnt it required that user 2 should also have a config?
I guess it tries to fetch the default config here right?